### ISSUE TITLE
508-defect-2 [COGNITION, FOCUS MANAGEMENT]: Focus order should preserve meaning

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#17448]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Jennifer

## User Story or Problem Statement

As a user with cognitive considerations, I expect to tab through the content and understand the content flow, so that there is a meaningful sequence. 

## Details

Tabbing from the search results to the map, the tab order is as follows:

1. canvas.mapboxgl-canvas
1. zoom in
1. zoom out
1. mapbox link
1. search this area of map div, with a button inside
1. link: Try using our other tools to search.
1. Then into the footer area links

The "Search this area of map" button should be before the Mapbox link. For low vision screen reader users, they may be disoriented. WCAG 2.0 Level A - 2.4.3 Focus Order states, "If a Web page can be navigated sequentially and the navigation sequences affect meaning or operation, focusable components receive focus in an order that preserves meaning and operability." Also, WCAG 2.0 Level A - 1.3.2 Meaningful Sequence states, "When the sequence in which content is presented affects its meaning, a correct reading sequence can be programmatically determined."

## Acceptance Criteria

- [ ] When a user tabs through the screen content, the "Search this area of map" button should be announced when it appears, and focus moved to it.
- [ ] If the user doesn't activate that button, the button should appear after the Zoom In/Out buttons for a meaningful reading order.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: any
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/find-locations/?address=Portland%2C%20Maine%2004103%2C%20United%20States&context=Portland%2C%20Maine%2004103%2C%20United%20States&facilityType=health&location=43.69%2C-70.30&page=1&searchString=Portland%2C%20Maine%2004103%2C%20United%20States&serviceType` in browser
2. Start tabbing through page content
3. Navigate to the map area
4. Zoom in or out
5. Verify that "Search this area of the map" is announced
6. Continue tabbing
7. Verify that the Mapbox link is the next in the flow, followed by the "Search this area of the map" button

## Solution (if known)

Reorder the order:

1. canvas.mapboxgl-canvas
1. zoom in
1. zoom out
1. search this area of map div, with a button inside
1. mapbox link
1. div.mapboxgl-ctrl mapboxgl-ctrl-attrib mapboxgl-compact
1. link: Try using our other tools to search.
1. Then into the footer area links
1. Feedback link (sitewide issue — [#17460](https://github.com/department-of-veterans-affairs/va.gov-team/issues/17460))

## WCAG or Vendor Guidance (optional)

* [WCAG 2.0 Level A - 2.4.3 Focus Order](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html) - If a Web page can be navigated sequentially and the navigation sequences affect meaning or operation, focusable components receive focus in an order that preserves meaning and operability.

* [WCAG 2.0 Level A - 1.3.2 Meaningful Sequence](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-sequence.html) - When the sequence in which content is presented affects its meaning, a correct reading sequence can be programmatically determined.

**Comment by mmiddaugh** [comment]
This issue will be resolved with the Mapbox Accessibility plugin immediately after the holidays

**Comment by mmiddaugh** [comment]
This is being addressed in #17665
